library/src/main/java/com/google/android/exoplayer/extractor/ts/Mpeg2Reader.java

```diff
@@ -0,0 +1,189 @@
+
+        Assertions.checkState(length >= 0);
+        if (length == 0) {
+            return false;
+        }
+
+        while (startoffset < length) {
+            if((data[startoffset + 0] & 0xFF) == 0x00 && (data[startoffset + 1] & 0xFF) == 0x00 && (data[startoffset + 2] & 0xFF) == 0x01 && (data[startoffset + 3] & 0xFF) == 0xB3) {
+                seqHeaderFound = true;
+                Log.i(TAG, "Found the Sequence Header");
+                return true;
+            }
+            startoffset++;
+            continue;
+        }
+        return false;
+    }
+
+    /**
+     * Locates the next frame start code (0x00, 0x00, 0x01, 0xXX)
+     * where XX can be anything lesser than 0x01 or more than 0xaf
+     * @param data The buffer whose position should be advanced.
+     * @param startoffset The offset of the data in {@code data}.
+     * @param endoffset The limit (exclusive) of the data in {@code data}.
+     * @return The offset into the buffer where a new frame starts from.
+     */
+    private int findValidStartCode(byte[] data, int startoffset, int endoffset) {
+        int length = endoffset - startoffset;
+
+        Assertions.checkState(length >= 0);
+        if (length == 0) {
+            return endoffset;
+        }
+
+        int limit = endoffset - 3;
+        for (int i = startoffset + 3; i < limit; i += 4) {
+            if ((data[i - 1] & 0xFE) != 0) {
+                // There isn't a frame prefix here, or at the next two positions. Do nothing and let the
+                // loop advance the index by four.
+            } else if (data[i - 3] == 0 && data[i - 2] == 0 && data[i - 1] == 1) {
+                if (checkMpegVideoFrame(data[i]))
+                   return i - 3;
+            } else {
+                i -= 3;
+            }
+        }
+        return endoffset;
+    }
+
+    public  boolean checkMpegVideoFrame (int data) {
+        int byte1 = data & 0xFF;
+        if (byte1 < 0x01 || byte1 > 0xaf){
+            isSync = (byte1 == 0xb3) ? true : false;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Parses the sample size from sequence header and prepares media format
+     * @param data The buffer whose position should be advanced.
+     * @param offset The offset of the data in {@code data}.
+     * @return None
+     */
+    private void findSampleSizeAndPrepareMediaFormat(byte[] data, int offset) {
+        if (seqHeaderFound) {
+            // Width and Height
+            int b1 = data[4 + offset] & 0xFF;
+            int b2 = data[5 + offset] & 0xFF;
+            int b3 = data[6 + offset] & 0xFF;
+            width = (b1 << 4) | (b2 & 0xF0);
+            height = (b2 & 0x0F) << 8 | b3;
+            Log.i(TAG, "Width is " + width + " Height is " + height);
+        }
+        List<byte[]> initializationData = new ArrayList<>();
+        output.format(MediaFormat.createVideoFormat(MediaFormat.NO_VALUE, MimeTypes.VIDEO_MPEG2,
```

Comment:
pixelWidthHeightRatio is not determined and passed, which causes wrong rendered video